sycl/include/sycl/device_aspect_macros.hpp

```diff
@@ -0,0 +1,409 @@
+//==------------------- device_aspect_macros.hpp - SYCL device -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __SYCL_ALL_DEVICES_HAVE_0__
+// __SYCL_ASPECT(host, 0)
+#define __SYCL_ALL_DEVICES_HAVE_0__ 0
+#endif
+
```

Comment:
Can we auto-generate it? Something like https://github.com/intel/llvm/blob/c4694cede8b2244e9c090e9131e70578ebfb6102/sycl/plugins/level_zero/CMakeLists.txt#L74-L83 maybe?

Reply:
We discussed earlier the possibility of auto-generation here: https://github.com/intel/llvm/pull/8392#discussion_r1113485607

Reply:
I personally strongly disagree with this, based on my experience with a similar issue of missing explicit instantiations for properties and the fact that new changes need to touch too many files.

However, it seems the team is in agreement with this so I won't block the PR.